app/controllers/concerns/sufia/users_controller_behavior.rb

```diff
@@ -142,4 +138,11 @@ def get_sort
            end
     return sort_val
   end
+
+  def deny_access(exception)
+    unless @user == current_user
```

Comment:
You're right, I'll change it.  Got ahead of myself by thinking about doing the same thing to user_not_current_user in which case deny_access would need a conditional for the two user cases.